include/class.file.php

```diff
@@ -162,7 +162,7 @@ function display($scale=false, $ttl=86400) {
 
         if ($scale && extension_loaded('gd')) {
             $image = imagecreatefromstring($this->getData());
-            $width = imagesx($image);
+            $width = $image ? imagesx($image) : 1;
```

Comment:
I think the fix here is incomplete - `imgesy` (line 167) would still expect an image. I think we should skip scalling  if  `imagecreatefromstring` fails.

```php

if ($scale  && extension_loaded('gd') 
          && ($image = imagecreatefromstring($this->getData()))) {